src/persist/src/indexed/mod.rs

```diff
@@ -523,4 +548,60 @@ mod tests {
         i.seal(id, 3)?;
         Ok(())
     }
+
+    #[test]
+    fn buf_snapshot_validate() {
```

Comment:
just want to flag that I named the validation function `validate_buf_snapshot` to match the `verb_noun` convention of `drain_buf` but I named this test `buf_snapshot_validate` to match the `noun_verb` convention of other validation tests. I think its fine but any suggestions for naming this are welcome

Reply:
In this case, I'd say just go for whatever names are the best combination of evocative and short. These are fine by me